packages/nextra/src/loader.ts

```diff
@@ -118,39 +113,29 @@ async function loader(
     },
     resourcePath
   )
+  const [pageMap, route, title] = getPageMap(
+    resourcePath,
+    pageMapResult,
+    fileMap,
+    defaultLocale
+  )
 
   if (unstable_flexsearch) {
-    if (data.searchable !== false) {
+    if (meta.searchable !== false) {
       addPage({
         fileLocale: fileLocale || DEFAULT_LOCALE,
         route,
         title,
-        data,
+        meta,
         structurizedData
       })
     }
 
     indexContentEmitted.add(filename)
   }
 
-  let timestamp: number | undefined
+  let timestamp: PageOpts['timestamp']
   if (repository && gitRoot) {
-    if (repository.isShallow() && !wasShallowWarningPrinted) {
-      if (process.env.VERCEL) {
-        console.warn(
-          '[nextra] The repository is shallow cloned, so the latest modified time will not be presented. Set the VERCEL_DEEP_CLONE=true environment variable to enable deep cloning.'
-        )
-      } else if (process.env.GITHUB_ACTION) {
-        console.warn(
-          '[nextra] The repository is shallow cloned, so the latest modified time will not be presented. See https://github.com/actions/checkout#fetch-all-history-for-all-tags-and-branches to fetch all the history.'
-        )
-      } else {
-        console.warn(
-          '[nextra] The repository is shallow cloned, so the latest modified time will not be presented.'
-        )
-      }
-      wasShallowWarningPrinted = true
-    }
```

Comment:
can be moved outside of loader (inside of IIFE) and `wasShallowWarningPrinted` variable can be removed